STM32/AC/Core/Src/ACBoard.c

```diff
@@ -156,9 +169,6 @@ static double ADCtoTemperature(double adc_val)
 WWDG_HandleTypeDef* hwwdg_ = NULL;
 static void printCurrentArray(int16_t *pData, int noOfChannels, int noOfSamples)
 {
-    // Update window watchdog. Will trigger reset outside window: <90ms && >110ms.
-    HAL_WWDG_Refresh(hwwdg_);
```

Comment:
So the iwdg crashes the program when storing the fault info in flash so it's removed. However, I don't see why we wouldn't keep on using the wwdg as an actual window wdg keeping the previous timer settings. 

The wwdg is being disabled when running flash so should be fine.

Reply:
Actually the IWDG isn't the problem. The IWDG can be set slow enough that it won't interfere with the flash operation

The problem seems to be that the print timing (and therefore watchdog reset) is driven off the ADC DMA. The DMA seems to be reset (or perhaps it keeps going, but the interrupt flag can't be set while flashing?) during the flash write operation.